src/main.ts

```diff
@@ -5,13 +5,26 @@ import deactivateEnvironment from "./deactivate";
 
 async function run() {
   try {
-    const { repo, ref, sha } = github.context;
+    const { repo: ctxRepo, ref, sha } = github.context;
+
+    /* 
+      this allows for `repository` to be
+      - null/undefined (defaults to current action repo)
+      - repo (defaults owner to same as current action)
+      - owner/repo 
+    */
+    const repositoryInput = core.getInput("repository");
+    let [
+      repository = ctxRepo.repo,
+      owner = ctxRepo.owner
+    ] = repositoryInput ? repositoryInput.split("/").reverse() : [];
```

Comment:
Would be good to log the evaluated result of this as well (ie a log message indicating which repo is being targeted, regardless of whether or not the `repository` field is set)